packages/material-ui/src/utils/useId.js

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+/**
+ * Private module reserved for @material-ui/x packages.
+ */
+export default function useId(idProp) {
+  const [defaultId, setDefaultId] = React.useState();
```

Comment:
As I began writing tests for this, I realised that it would always issue a second render, even if `idProp` is provided. @oliviertassinari what do you think about this diff?

```suggestion
  const [defaultId, setDefaultId] = React.useState(idProp);
```